packages/editor/src/client/panel/memberPlanEditPanel.tsx

```diff
@@ -0,0 +1,361 @@
+      )
+      setIsActive(data.memberPlan.isActive)
+      setAvailablePaymentMethods(
+        data.memberPlan.availablePaymentMethods
+          ? data.memberPlan.availablePaymentMethods.map(availablePaymentMethod => ({
+              id: generateID(),
+              value: availablePaymentMethod
+            }))
+          : []
+      )
+      setPricePerMonthMinimum(data.memberPlan.pricePerMonthMinimum)
+      setPricePerMonthMaximum(data.memberPlan.pricePerMonthMaximum)
+    }
+  }, [data?.memberPlan])
+
+  useEffect(() => {
+    if (paymentMethodData?.paymentMethods) {
+      setPaymentMethods(paymentMethodData.paymentMethods)
+    }
+  }, [paymentMethodData?.paymentMethods])
+
+  useEffect(() => {
+    const error =
+      loadError?.message ??
+      createError?.message ??
+      updateError?.message ??
+      paymentMethodLoadError?.message
+    if (error) Alert.error(error, 0)
+  }, [loadError, createError, updateError, paymentMethodLoadError])
+
+  function handleImageChange(image: ImageRefFragment) {
+    setImage(image)
+  }
+
+  async function handleSave() {
+    if (id) {
+      const {data} = await updateMemberPlan({
+        variables: {
+          id,
+          input: {
+            name,
+            imageID: image?.id,
+            description,
+            isActive,
+            availablePaymentMethods: availablePaymentMethods.map(({value}) => ({
+              paymentPeriodicities: value.paymentPeriodicities,
+              forceAutoRenewal: value.forceAutoRenewal,
+              paymentMethodIDs: value.paymentMethods.map(pm => pm.id)
+            })),
+            pricePerMonthMinimum,
+            pricePerMonthMaximum: fixPrice ? pricePerMonthMinimum : pricePerMonthMaximum
+          }
+        }
+      })
+
+      if (data?.updateMemberPlan) onSave?.(data.updateMemberPlan)
+    } else {
+      const {data} = await createMemberPlan({
+        variables: {
+          input: {
+            name,
+            imageID: image?.id,
+            description,
+            isActive,
+            availablePaymentMethods: availablePaymentMethods.map(({value}) => ({
+              paymentPeriodicities: value.paymentPeriodicities,
+              forceAutoRenewal: value.forceAutoRenewal,
+              paymentMethodIDs: value.paymentMethods.map(pm => pm.id)
+            })),
+            pricePerMonthMinimum,
+            pricePerMonthMaximum: fixPrice ? pricePerMonthMinimum : pricePerMonthMaximum
+          }
+        }
+      })
+
+      if (data?.createMemberPlan) onSave?.(data.createMemberPlan)
+    }
+  }
+
+  return (
+    <>
+      <Drawer.Header>
+        <Drawer.Title>
+          {id ? t('memberPlanList.editTitle') : t('memberPlanList.createTitle')}
+        </Drawer.Title>
+      </Drawer.Header>
+
+      <Drawer.Body>
+        <Panel>
+          <Form fluid={true}>
+            <FormGroup>
+              <ControlLabel>{t('memberPlanList.name')}</ControlLabel>
```

Comment:
I would maybe consider splitting those translations up into what you have in the overview (so `memberplan.overview.XXX`) and in a drawer/panel (`memerPlan.panels.XXX`)  at least that's the structure I've used so far.

Reply:
True. But this whole thing will change a lot over the coming week and then this will be fixed.